src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphAuthDAO.scala

```diff
@@ -74,7 +74,7 @@ class GraphAuthDAO extends AuthDAO with GraphDAO {
   }
 
   override def getMaximumAccessLevel(user: RawlsUserRef, workspaceId: String, txn: RawlsTransaction): WorkspaceAccessLevel = {
-    val workspaces = listWorkspaces(user, isTargetWorkspace(workspaceId), txn)
+    val workspaces = getWorkspaceAccessLevels(user, isTargetWorkspace(workspaceId), txn)
```

Comment:
Cheeky, but legit. Make sure it's tidy on the SQL side when the time comes,
though!
On Feb 29, 2016 9:18 AM, "MatthewBemis" <notifications@github.com> wrote:

> In
> src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphAuthDAO.scala
> <https://github.com/broadinstitute/rawls/pull/316#discussion_r54414502>:
>
> > @@ -74,7 +74,7 @@ class GraphAuthDAO extends AuthDAO with GraphDAO {
> >    }
> >
> >    override def getMaximumAccessLevel(user: RawlsUserRef, workspaceId: String, txn: RawlsTransaction): WorkspaceAccessLevel = {
> > -    val workspaces = listWorkspaces(user, isTargetWorkspace(workspaceId), txn)
> > +    val workspaces = getWorkspaceAccessLevels(user, isTargetWorkspace(workspaceId), txn)
>
> @helgridly <https://github.com/helgridly> The cause for the CI failures
> was that getMaximumAccessLevel used listWorkspaces, which caused all of the
> code that checked workspace permissions to break (since it would return
> NoAccess if not in the realm). getMaximumAccessLevel should NOT use this
> new logic, so instead of spending time re-implementing everything I took
> the lazy approach and just copied the old version of listWorkspaces into a
> new function called getWorkspaceAccessLevels. Since we're destroying all of
> this code within the next few weeks anyway, I argue it isn't worth my time
> to further re-work this.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/broadinstitute/rawls/pull/316/files#r54414502>.
>